client/src/pages/List.js

```diff
@@ -14,72 +16,191 @@ const useStyles = makeStyles({
+          items: sourceItems,
+        },
+        [destination.droppableId]: {
+          ...destinationColumn,
+          items: destinationItems,
+        },
+      });
+    } else {
+      const column = columns[source.droppableId];
+      const copiedItems = [...column.items];
+      const [removed] = copiedItems.splice(source.index, 1);
+      copiedItems.splice(destination.index, 0, removed);
+      setColumns({
+        ...columns,
+        [source.droppableId]: {
+          ...column,
+          items: copiedItems,
+        },
+      });
+    }
   }
 
   return (
     <div className={classes.mainContainer}>
-      <Typography variant="h1">List</Typography>
-      <DragDropContext onDragEnd={handleOnDragEnd}>
-        <Droppable droppableId="column">
-          {(provided) => (
-            <Grid
-              className={classes.column}
-              container
-              direction="column"
-              justify="center"
-              {...provided.droppableProps}
-              ref={provided.innerRef}>
-              {list.map((e, idx) => (
-                <Draggable key={e} draggableId={e} index={idx}>
+      <DragDropContext
+        onDragEnd={(result) => handleOnDragEnd(result, columns, setColumns)}>
+        <div className={classes.columnsContainer}>
+          {Object.entries(columns).map(([id, column]) => {
+            return (
+              <div>
+                <Droppable droppableId={id}>
                   {(provided) => (
-                    <Card
-                      className={classes.card}
-                      {...provided.draggableProps}
-                      {...provided.dragHandleProps}
+                    <Grid
+                      className={classes.column}
+                      container
+                      direction="column"
+                      justify="flex-start"
+                      {...provided.droppableProps}
                       ref={provided.innerRef}>
-                      {e}
-                    </Card>
+                      <div className={classes.cardHeader}>
+                        <Typography variant="h5">{column.name}</Typography>
+                        <MoreHoriz className={classes.moreHoriz} />
+                      </div>
+                      {column.items.map((e, idx) => (
+                        <Draggable
+                          key={`${idx} ${e.title}`}
+                          draggableId={`${idx} ${e.title}`}
+                          index={idx}>
+                          {(provided) => (
+                            <Card
+                              className={classes.card}
+                              {...provided.draggableProps}
+                              {...provided.dragHandleProps}
+                              ref={provided.innerRef}>
+                              <span
+                                className={classes.cardStatus}
+                                style={{ backgroundColor: e.status }}></span>
```

Comment:
Would be better to make the status an additional class that specifies the color, easier to edit in the future. You can use two classes by joining the names in a string

Reply:
That's a great idea, thanks!